bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/protocol/http/WebServerApi.java

```diff
@@ -0,0 +1,558 @@
+
+    private static final String REQ_HDR_APPLICATION_JSON = new MediaType("application", "json", "UTF-8").toString();
+
+    private WebServerApi() {
+    }
+
+    public static WebServerApi getInstance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Sets the httpClient object to be used for API calls to Vesync.
+     *
+     * @param httpClient the client to be used.
+     */
+    public void setHttpClient(@Nullable HttpClient httpClient) {
+        if (httpClient != null) {
+            this.httpClient = httpClient;
+        }
+    }
+
+    public Map<String, String> getBridgeProperities(final String hostname)
+            throws NotTapLinkGatewayException, TransientCommunicationIssueException {
+        try {
+            final Request request = httpClient.newRequest(URI_HOST_PREFIX + hostname).method(HttpMethod.GET);
+            final ContentResponse cr = request.timeout(REQ_TIMEOUT_SECONDS, TimeUnit.SECONDS).send();
+            if (HttpURLConnection.HTTP_OK != cr.getStatus()) {
+                throw new NotTapLinkGatewayException(UNEXPECTED_STATUS_CODE);
+            }
+            validateHeaders(cr.getHeaders());
+            final String responseData = cr.getContentAsString();
+            final Document doc = Jsoup.parse(responseData);
+
+            switch (doc.title()) {
+                case TITLE_API_CONFIG_PAGE:
+                    break;
+                case TITLE_API_LOGIN_PAGE:
+                    return Map.of();
+                default:
+                    throw new NotTapLinkGatewayException(MISSING_SERVER_TITLE);
+            }
+            final Map<String, String> deviceProps = getMetadataProperties(doc);
+            LinkTapBridgeHandler.Firmware firmware = new LinkTapBridgeHandler.Firmware(
+                    deviceProps.get(BRIDGE_PROP_GW_VER));
+            if (firmware.supportsMDNS()) {
+                getMdnsEnableArgs(doc);
+            } else {
+                logger.debug("Firmware revision does not include mDNS support");
+            }
+            return deviceProps;
+
+        } catch (InterruptedException e) {
+            return Map.of();
+        } catch (TimeoutException e) {
+            throw new TransientCommunicationIssueException(HOST_COMM_TIMEOUT);
+        } catch (ExecutionException e) {
+            final Throwable t = e.getCause();
+            if (t instanceof UnknownHostException) {
+                throw new TransientCommunicationIssueException(HOST_NOT_RESOLVED);
+            } else if (t instanceof SocketTimeoutException) {
+                throw new TransientCommunicationIssueException(HOST_UNREACHABLE);
+            } else if (t instanceof SSLHandshakeException) {
+                throw new NotTapLinkGatewayException(UNEXPECTED_HTTPS);
+            } else {
+                logger.warn("ExecutionException -> {}", e.getMessage());
+            }
+            throw new NotTapLinkGatewayException("Unexpected failure -> " + e.getMessage());
```

Comment:
You might create another specific exception or rename this to a `LinkTapGatewayException` and differentiate with the error message.
Currenlty the `NotTapLinkGatewayException` and related javadoc indicate the device is not a linktap device. But an SSL error (see above) or this 'unknown' error do not clearly relate to the device NOT being a LinkTap device. 
(hope this makes sense)

Reply:
So this ties back to the other bit. The gateways don't have SSL basically. So I was trying different ways, to ensure it didn't continue with incorrect addresses. In this case, I believe the client followed a redirect to a https Nginx version of a different system when I entered the IP of a different server here. So because as LinkTap does not have SSL support if there is any SSL going on it cannot be either the v1 or v2 gateway. That's why it defaults currently to this exception, rather than saying there is a security related exception. Hopefully that makes sense?

Reply:
This makes sense for the catch block that catches the `SSLException`.
Line 141 (this comment) is a generic/unknown failure, why is that also linked to `NotTapLinkGatewayException`

Reply:
Yeah good question - I'll have to go back through the stack and try to remember - may need fixing, or other failures thrown also fit like the SSL one - will cycle back to this one.

Reply:
I had a look through this. Basically, the others catch what could be the expected failures. This is a catch all I believe, as the happy path generally never fail's, but if there is something unexpected that reaches this point it should be because the device is not a tap link device. Hence the fall back to this exception.

Reply:
Still a big step to call it not a linktap device by convention. This is an unknown situation/path and that should be reflected in the exception. I would still like this to changed to a generic Linktapexception

Reply:
Sure that makes sense, its been updated to a generic LinktapException like suggested.

Reply:
@lsiepel Hopefully this last push closes off the last comment as far as I can tell 🤞 If it does thank you so much for all the time and effort in reviewing, and pointers during the review. I will have to re-apply some of the changes to that other PR.

Reply:
Looks good, did you see my last comment at the bottom  about aditional testing before merging?